watch-library/hardware/watch/watch_adc.c

```diff
@@ -138,13 +139,30 @@ void watch_set_analog_sampling_length(uint8_t cycles) {
     _watch_sync_adc();
 }
 
+static inline uint32_t _watch_adc_get_reference_voltage(const watch_adc_reference_voltage reference) {
+    switch (reference) {
+        case ADC_REFERENCE_INTREF:
+            return ADC_REFCTRL_REFSEL_INTREF_Val;
+            break;
+        case ADC_REFERENCE_VCC_DIV1POINT6:
+            return ADC_REFCTRL_REFSEL_INTVCC0_Val;
+            break;
+        case ADC_REFERENCE_VCC_DIV2:
+            return ADC_REFCTRL_REFSEL_INTVCC1_Val;
+            break;
+        case ADC_REFERENCE_VCC:
+            return ADC_REFCTRL_REFSEL_INTVCC2_Val;
+            break;
+    }
+}
```

Comment:
This could probably be a lookup table since the enum is an integer and these values are contiguous.

Reply:
Yep yep, but for now this is clear — and moving it to an enum matches with how the watch library tends to define other things of this nature.